### DISCUSSION TITLE
Each channel -- Left and Right -- on an independent host?

### DISCUSSION THREAD

**Discussion by twork** [MusicPlayerDaemon/MPD#1769] [General]
I'm new to MPD, and need some tips on where I should be reading docs.  I think my scheme will work, but might be looking at it the wrong way.

I have a room where WiFi is the only way I can spread hosts around.  My plan is to have three Linux Pi hosts: one with the data stored, then the "Left Pi", and the "Right Pi" hosts, each with one speaker attached.

First question of course: is this doable using MPD as the basis?  Is it a common approach, "go look at [web page]"?  Or, doable, but not common so [issues]?

I found on the MPD docs:

  https://mpd.readthedocs.io/en/stable/user.html#real-time-scheduling

...that looks promising, and maybe it answers my question, but I still don't know how to go about my project.

Hints?  Thanks.

**Comment by twork** [comment]
I should add: it's not clear to me, on the three hosts, where instances of MPD should run.  Does it emit sound data at the central/storage host, which is then piped out of a sound app at each speaker-side host?

Or, does MPD act in all three hosts, and that's how we guarantee that output is kept in sync?  In that case my guess would still be that there is some other app which translates the data into sound on the speaker, I'm just not clear on how the activity is handled to stay in sync.

Thanks.

**Comment by skidoo23** [comment]
Real time scheduling does not help you. May I quote the docs : "There is a rumor that real-time scheduling improves audio quality. That is not true. All it does is reduce the probability of skipping (audio buffer xruns) when the computer is under heavy load."

Use Snapcast on your central MPD and Snapcast commandline clients on "left" and "right" raspberries. But... I have now idea how seperation of left and right channel impacts sound quality. Better use wires. Really.

**Comment by twork** [comment]
Thanks.

To be clear though, I don't have any belief that realtime scheduling
improves anything up close. Only that it tries to guarantee that a
scheduled event executed on two hosts will happen at the same time. And,
there is work underway for this sort of thing, by others, in more important
undertakings.  So I figured...

For whatever it's worth, I also hope to eventually play some experimental
music composed in the 1960's, requiring more than two channels.  This was
in some sense a test case.  I was prepared to be told "nope," but it's a
bummer.

Csound next I guess. If it can't do this...

**Comment by kingosticks** [comment]
> Only that it tries to guarantee that a
scheduled event executed on two hosts will happen at the same time

Pretty sure it doesn't do that.

You are best off with Snapcast. There's probably some way (outside of the Snapcast client) to discard the channel you don't need on each side.
